.github/workflows/build_test.yaml

```diff
@@ -0,0 +1,29 @@
+name: Nosetests
+on: [push]
+
+jobs:
+  build:
+    name: Validate Changes
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [2.7, 3.8]
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Fetch master (to enable diff)
+      run: |
+        git fetch origin master
```

Comment:
actions/checkout has a fetch-depth option that we could use here. A value of '0' would give us a full checkout but '2-N' for some small N might do exactly what we need depending on the behavior of PR setups.

Reply:
I looked through the docs and if you want to fetch other branches they point to calling fetch directly. but switching to the shallow fetch with `--depth` cuts it down to a zero second fetch since it's typically just a ref just behind the existing PR.